src/settings/NoticePolicy/NoticePolicyDetail.test.js

```diff
@@ -0,0 +1,185 @@
+import React from 'react';
+import {
+  render,
+  screen,
+  fireEvent,
+} from '@testing-library/react';
+
+import '../../../test/jest/__mock__';
+
+import {
+  ExpandAllButton,
+  AccordionSet,
+} from '@folio/stripes/components';
+
+import NoticePolicyDetail from './NoticePolicyDetail';
+import {
+  GeneralSection,
+  LoanNoticesSection,
+  RequestNoticesSection,
+  FeeFineNoticesSection,
+} from './components/DetailSections';
+import { NoticePolicy } from '../Models/NoticePolicy';
+import { patronNoticeCategoryIds } from '../../constants';
+
+jest.mock('./components/DetailSections', () => ({
+  GeneralSection: jest.fn(() => null),
+  LoanNoticesSection: jest.fn(() => null),
+  RequestNoticesSection: jest.fn(() => null),
+  FeeFineNoticesSection: jest.fn(() => null),
+}));
+jest.mock('./utils/get-templates', () => jest.fn((templates, categoryIds) => (
+  {
+    templates,
+    categoryIds,
+  }
+)));
+
+ExpandAllButton.mockImplementation(jest.fn(({ onToggle, ...rest }) => {
+  const sections = {
+    generalNoticePolicy: false,
+    viewLoanNotices: false,
+    viewRequestNotices: false,
+    viewFeeFineNotices: false,
+  };
+  return (
+    <button
+      onClick={() => onToggle(sections)}
+      type="button"
+      {...rest}
+    >
+        Toggle accordion state
+    </button>
+  );
+}));
+AccordionSet.mockImplementation(jest.fn(({ children, onToggle, ...rest }) => {
+  return (
+    // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
+    <div
+      onClick={() => onToggle({ id: 'generalNoticePolicy' })}
+      {...rest}
```

Comment:
same here. I would discuss with you and Dmitry this point

Reply:
Refactored, thanks!